lib/client/weblogin_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHostCredentialsFallback(t *testing.T) {
```

Comment:
Same as I'd other test, I'd consider making this one a table-driven test focused on HTTP fallback, with HostCredentials treated simply as one of the funcs being tested. Makes it easier to expand the test if, (god forbid), we add fallbacks to more endpoints.

Reply:
Done